editor/debugger/script_editor_debugger.cpp

```diff
@@ -2002,6 +2005,13 @@ ScriptEditorDebugger::ScriptEditorDebugger() {
 		add_child(file_dialog);
 	}
 
+	{ // expression evaluator
+		expression_evaluator = memnew(EditorExpressionEvaluator);
+		expression_evaluator->set_name(TTR("Evaluator"));
+		expression_evaluator->set_editor_debugger(this);
```

Comment:
This is a bit weird, other debugger components (i.e. non-plugins) use signals to communicate with the debugger (which does the sending internally)

Reply:
Something like

```suggestion
		expression_evaluator->connect("evaluate_expression", callable_mp(this,&ScriptEditorDebugger::_evaluate_expression));
```

Reply:
I think it is good because there are other things too that the evaluator checks with the ScriptEditorDebugger instance https://github.com/rohanrhu/godot/blob/a971a441dc3e0c5f08a3774e3048fda550ed54bd/editor/debugger/editor_expression_evaluator.cpp#L56

I can change all of those things and make it work with signals but I think this way is more efficient in terms of optimization.